pkg/tenant/api/v1/errors.go

```diff
@@ -28,6 +29,36 @@ var (
 	ErrUnparsable             = errors.New("could not parse request")
 )
 
+// FieldValidationError represents a validation error for a specific field, when the
+// frontend needs to know which field is in error, and includes the name of the field
+// and the index if the field is a list along with the error string.
+type FieldValidationError struct {
+	Field string `json:"field"`
+	Err   string `json:"error"`
+	Index int    `json:"index"`
+}
+
+func (e *FieldValidationError) Error() string {
+	switch {
+	case e.Field == "":
+		return e.Err
+	case e.Index > -1:
+		return fmt.Sprintf("%q at index %d: %s", e.Field, e.Index, e.Err)
+	default:
+		return fmt.Sprintf("%q: %s", e.Field, e.Err)
+	}
+}
+
+type FieldValidationErrors []*FieldValidationError
```

Comment:
I was thinking we could add this to the standard `Reply` struct for 400 errors and have `json:"omitempty"` for this and the `Error` field so that it would be one or the other in responses.

Reply:
Yes, that makes sense to me.